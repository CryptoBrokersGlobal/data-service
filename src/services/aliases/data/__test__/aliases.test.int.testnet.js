const { Nothing } = require('folktale/maybe');

const { createPgDriver } = require('../../../../db');
const loadConfig = require('../../../../loadConfig');
const options = loadConfig();
const create = require('../index');

const ADDRESS = '3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek';

describe('Aliases', () => {
  const service = create({
    pg: createPgDriver(options),
    emitEvent: () => () => null,
  });

  it('should return Maybe(alias) for one correctly', done => {
    service
      .get('test-alias')
      .run()
      .listen({
        onResolved: maybeX => {
          expect(maybeX).toMatchSnapshot();
          done();
        },
      });

    service
      .get('NON_EXISTING_ALIAS')
      .run()
      .listen({
        onResolved: maybeX => {
          expect(maybeX).toEqual(Nothing());
          done();
        },
      });
  });

  describe('request by address', () => {
    it('should return correct data if requested without showBroken', done => {
      service
        .mget({ address: ADDRESS })
        .run()
        .listen({
          onResolved: mxs => {
            expect(mxs).toMatchSnapshot();
            done();
          },
        });
    });

    it('should return correct data if requested with showBroken', done => {
      service
        .mget({
          address: ADDRESS,
          showBroken: true,
        })
        .run()
        .listen({
          onResolved: mxs => {
            expect(mxs).toMatchSnapshot();
            done();
          },
        });
    });
  });
});
