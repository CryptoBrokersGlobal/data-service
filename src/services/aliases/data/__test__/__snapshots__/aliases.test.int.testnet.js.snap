// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aliases request by address should return correct data if requested with showBroken 1`] = `
Array [
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test-alias",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test-alias-2",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test-alias-3",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test-alias-4",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test-alias-5",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test-alias-6",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test-alias-7",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test@test.test",
      },
    },
  },
]
`;

exports[`Aliases request by address should return correct data if requested without showBroken 1`] = `
Array [
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test-alias",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test-alias-2",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test-alias-3",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test-alias-4",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test-alias-5",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test-alias-6",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test-alias-7",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test@test.test",
      },
    },
  },
]
`;

exports[`Aliases should return Maybe(alias) for one correctly 1`] = `
Object {
  "@@tag": "Nothing",
  "@@type": "folktale:Maybe",
  "@@value": Object {},
}
`;
